src/udp_wrap.cc

```diff
@@ -213,6 +214,25 @@ X(SetMulticastLoopback, uv_udp_set_multicast_loop)
 
 #undef X
 
+void UDPWrap::SetMulticastInterface(const FunctionCallbackInfo<Value>& args) {
+  UDPWrap* wrap;
+  ASSIGN_OR_RETURN_UNWRAP(&wrap,
+                          args.Holder(),
+                          args.GetReturnValue().Set(UV_EBADF));
+
+  CHECK_EQ(args.Length(), 1);
+
+  node::Utf8Value iface(args.GetIsolate(), args[0]);
+
+  const char* iface_cstr = *iface;
+  if (args[0]->IsUndefined() || args[0]->IsNull()) {
```

Comment:
> args[0]->IsUndefined() seems to be untouchable because we checked the address at the JS level.

I think the JS checks are just a courtesy to let programs that don't do anything seriously wrong deal with an exception and live, I can bypass it and start triggering these paths, i,e:

      socket.on('listening', () => {
          socket._handle.setMulticastInterface();

so to verify there's no unaccountable corruption, I think this side needs to force an abort whenever it hasn't verified that the caller has been railroaded into a correct path.

How about:

       CHECK_EQ(args.Length(), 1);
       CHECK(args[0]->IsString());
       Utf8Value iface(args.GetIsolate(), args[0]);
       int err = uv_udp_set_multicast_interface(&wrap->handle_, *iface);